### ISSUE TITLE
__asan_handle_no_return causes crash

### ISSUE THREAD

**Issue by keitee** [google/sanitizers#1020]
Hi all,

I have managed to get crash in this function and have found that GetCurrentThread() returns null. If I comment out from this check and the rest then it would run further but system eventually hangs. 

void NOINLINE __asan_handle_no_return() {
  int local_stack;
  AsanThread *curr_thread = GetCurrentThread();
  CHECK(curr_thread);
  // skipped
}

I see [https://github.com/google/sanitizers/issues/189](url) is related but not sure what it means. Could anyone shed some light on this for me please?

1. When and why this __asan_handle_no_return gets called?
2. Is there any suggestion to get over this? Like to see if can continue on or have to give up to use Asan on this system. 

The env details are:
i686 target
CT_BINUTILS_VERSION="2.28"
CT_KERNEL_VERSION="3.14.43"
CT_LIBC_VERSION="2.25"
CT_CC_GCC_VERSION="4.9.4"

Many thanks for any comments.

**Comment by ramosian-glider** [comment]
It would also help to see some kind of a reproducer for the problem you're seeing.

**Comment by keitee** [comment]
Thanks for a reply. If you mean the way to reproduce, that's difficult since it's rather big system and don't have a test case/ sample application to reproduce. If knew why this happens, can try to find the use case from this system and to make a sample out of it.

**Comment by kcc** [comment]
Does this mean you are using asan with gcc 4.9.4? It's pretty old. 
Does this happen with the fresh clang?

**Comment by keitee** [comment]
Yes, that's old as the system is and have tried gcc 6.3.0 as well but no luck. In order to try clang, two problems:

1. need to have clang cross toolchain
2. need to build whole system with clang which may not a plain sailing. 

Yes, will give it a try to do 1 first.

**Comment by eugenis** [comment]
This could help:
https://llvm.org/docs/GettingStarted.html#getting-a-modern-host-c-toolchain
